pkg/controller/persistentvolume/controller_base.go

```diff
@@ -139,14 +139,14 @@ func (ctrl *PersistentVolumeController) initializeCaches(volumeSource, claimSour
 		glog.Errorf("PersistentVolumeController can't initialize caches: %v", err)
 		return
 	}
-	volumeList, ok := volumeListObj.(*api.List)
+	volumeList, ok := volumeListObj.(*api.PersistentVolumeList)
```

Comment:
I did try this fix first, but it made a mess of the tests. +1 if you can get the unit tests working. I'll try a bit more here ...